katas/pokerhands/percy/pokerHand.js

```diff
@@ -0,0 +1,465 @@
+            },
+            value:{
+                numberTwo:1,
+                numberThree:2,
+                numberFour:3,
+                numberFive:4,
+                numberSix:5,
+                numberSeven:6,
+                numberEight:7,
+                numberNine:8,
+                numberT:9,
+                numberJ:10,
+                numberQ:11,
+                numberK:12,
+                numberA:13
+            },
+            suit:{
+                diamonds:'D',
+                heart:'H',
+                spades:'S',
+                clubs:'C'
+            }
+
+        };
+
+
+        //four of a kind
+        if (this.#winningMove == undefined) {
+            if (this.objectValues.howMany.numberA == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberK == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberQ == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberJ == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberT == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberNine == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberEight == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberSeven == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberSix == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberFive == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberFour == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberThree == 4) {
+                this.#winningMove = 'four of a kind';
+            } else if (this.objectValues.howMany.numberTwo == 4) {
+                this.#winningMove = 'four of a kind';
+            }
+        }
+
+        //full house
+        if (this.#winningMove == undefined) {
+            if (this.objectValues.howMany.numberA == 3 && (this.objectValues.howMany.numberTwo == 2 || this.objectValues.howMany.numberThree == 2 || this.objectValues.howMany.numberFour == 2 || this.objectValues.howMany.numberFive == 2 || this.objectValues.howMany.numberSix == 2 || this.objectValues.howMany.numberSeven == 2 || this.objectValues.howMany.numberEight == 2 || this.objectValues.howMany.numberNine == 2 || this.objectValues.howMany.numberT == 2 || this.objectValues.howMany.numberJ == 2 || this.objectValues.howMany.numberQ == 2 || this.objectValues.howMany.numberK == 2)) {
```

Comment:
These conditions are very complex, it would be good to think an alternative where this logic re-use conditions or use separated functions so they are more readable and less prone to error, a bug in this code would be challenging to find/solve